junit-jupiter-api/src/main/java/org/junit/jupiter/api/Assertions.java

```diff
@@ -54,6 +54,9 @@ private Assertions() {
 	 * <p>See Javadoc for {@link #fail(String, Throwable)} for an explanation of
 	 * this method's generic return type {@code V}.
 	 */
+	// we purposely return a V that has nothing to do with the parameters of this method; see
+	// the javadoc for details.
+	@SuppressWarnings("TypeParameterUnusedInFormals")
```

Comment:
Unfortunately, I don't think there is a more elegant way.

AFAICT, it's the only solution that the [docs](http://errorprone.info/bugpattern/TypeParameterUnusedInFormals) for that rule provides.

(The docs for all the other rules I've looked at in the past suggest the exact same workaround, IIRC)